### PR TITLE
issue #175 : upgrading the dependencies

### PR DESCRIPTION
**Pull request by superiorsd10**
#175 
I've updated the dependencies to the latest version without any dependencies issues constraints. There is some dependency that I've not updated because another dependency depends on their lower versions. So it'll create a major problem in version solving.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -35,25 +35,21 @@ dependencies:
...
   # Implementation of the LIFX HTTP API
   lifx_http_api: ^0.0.13
   # Small, easy to use and extensible logger which prints beautiful logs.
   logger: ^1.1.0
   # A server and browser based MQTT client for Dart.
-  mqtt_client: ^9.7.1
+  mqtt_client: ^9.7.2
   # Service discovery over multicast DNS (mDNS), Bonjour, and Avahi.
-  multicast_dns: ^0.3.2+2
+  multicast_dns: ^0.3.2+1
```
Comment (guyluz11) [inline pubspec.yaml:50]: Why downgrade `multicast_dns` package version?

File: pubspec.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ homepage: https://cybearjinni.com
 publish_to: none
 
 environment:
-  sdk: ^2.18.0
+  sdk: ^2.17.1
```
Comment (guyluz11) [inline pubspec.yaml:8]: Why downgrade the project dart version?.

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Left some requests for changes, please review.

**Comment by superiorsd10** [comment]
@guyluz11 I've reviewed the changes. Please look into this now.

**Review by guyluz11 — APPROVED** [review]

**Pull request merged**